crates/router/src/connector/forte.rs

```diff
@@ -0,0 +1,644 @@
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        // todo!()
+        Ok(self.base_url(_connectors).to_string())
+    }
+
+    fn get_request_body(
+        &self,
+        _req: &types::PaymentsCaptureRouterData,
+    ) -> CustomResult<Option<String>, errors::ConnectorError> {
+        // todo!()
+        let connector_req = forte::FortePaymentsCaptureRequest::try_from(_req)?;
+        let forte_req = utils::Encode::<forte::FortePaymentsCaptureRequest>::encode_to_string_of_json(
+            &connector_req,
+        )
+        .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(Some(forte_req))
+        
+    }
+
+    fn build_request(
+        &self,
+        req: &types::PaymentsCaptureRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        Ok(Some(
+            services::RequestBuilder::new()
+                .method(services::Method::Put)
+                .url(&types::PaymentsCaptureType::get_url(self, req, connectors)?)
+                .headers(types::PaymentsCaptureType::get_headers(
+                    self, req, connectors,
+                )?)
+                .body(types::PaymentsCaptureType::get_request_body(self, req)?)
+                .build(),
+        ))
+    }
+
+    fn handle_response(
+        &self,
+        data: &types::PaymentsCaptureRouterData,
+        res: Response,
+    ) -> CustomResult<types::PaymentsCaptureRouterData, errors::ConnectorError> {
+
+        // let type_a = Type_A::try_from(type_b)?;
+
+        // let type_a: Type_A = type_b.try_into()?;
+
+
+        let response: forte::ForteResponse = res
+            .response
+            .parse_struct("Forte Response")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        // println!("something3 {}",response);
+        logger::debug!(forte_create_response=?response);
+        types::RouterData::try_from(types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        })
+        // types::ResponseRouterData {
+        //     response,
+        //     data: data.clone(),
+        //     http_code: res.status_code,
+        // }
+        // .try_into()
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
+    }
+
+    fn get_error_response(
+        &self,
+        res: Response,
+    ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
+        let response: forte::ErrorDesc = res
+            .response
+            .parse_struct("Forte ErrorDesc")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        Ok(ErrorResponse {
+            status_code: res.status_code,
+            code: res.status_code.to_string(),
+            message: response.response.response_desc,
+            reason: None,
+        })
```

Comment:
```suggestion
        self.build_error_response(res)
```
move this to build_error_response and use self.build_error_response(res)